common/src/test/kotlin/io/homeassistant/companion/android/common/data/network/NetworkStatusMonitorImplTest.kt

```diff
@@ -0,0 +1,153 @@
+package io.homeassistant.companion.android.common.data.network
+
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkRequest
+import io.homeassistant.companion.android.database.server.ServerConnectionInfo
+import io.mockk.CapturingSlot
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class NetworkStatusMonitorImplTest {
+
+    private val connectivityManager: ConnectivityManager = mockk()
+    private val networkHelper: NetworkHelper = mockk()
+    private val network: Network = mockk()
+    private val serverConfig: ServerConnectionInfo = mockk(relaxed = true)
+
+    private lateinit var networkMonitor: NetworkStatusMonitor
+    private lateinit var callbackSlot: CapturingSlot<ConnectivityManager.NetworkCallback>
+
+    @BeforeEach
+    fun setup() {
+        callbackSlot = slot()
+        networkMonitor = NetworkStatusMonitorImpl(connectivityManager, networkHelper)
+
+        every {
+            connectivityManager.registerNetworkCallback(
+                any<NetworkRequest>(),
+                capture(callbackSlot),
+            )
+        } just Runs
+
+        every {
+            connectivityManager.unregisterNetworkCallback(any<ConnectivityManager.NetworkCallback>())
+        } just Runs
+    }
+
+    @Test
+    fun `given monitoring flow when monitoring is canceled then flow unregisters callback`() = runTest {
+        every { networkHelper.hasActiveNetwork() } returns true
+        every { networkHelper.isNetworkValidated() } returns true
+        val job = launch {
+            networkMonitor.observeNetworkStatus(serverConfig).collect()
+        }
+        advanceUntilIdle()
+        job.cancel()
+        advanceUntilIdle()
+
+        verify { connectivityManager.unregisterNetworkCallback(callbackSlot.captured) }
+    }
+
+    @Test
+    fun `given no active network when monitoring starts then emit UNAVAILABLE`() = runTest {
```

Comment:
```suggestion
    fun `given no active network when monitoring starts then emits UNAVAILABLE`() = runTest {
```
To confirm with @jpelgrom my english grammar is far from being accurate ...

Reply:
Different suggestion to avoid the discussion and for consistency with the final two tests:

```suggestion
    fun `given no active network when monitoring starts then state is UNAVAILABLE`() = runTest {
```

FWIW I think emits is correct, but you're missing the actor in the title (_who/what_ emits?). emit is what the function will do (via `trySend`) but that's not what you're testing.